libminifi/test/unit/MinifiConcurrentQueueTests.cpp

```diff
@@ -123,37 +234,79 @@ TEST_CASE("TestConqurrentQueue::testQueueWithReAdd", "[TestQueueWithReAdd]") {
-    queue.enqueue("dum");
-    std::this_thread::sleep_for(std::chrono::milliseconds(3));
-    queue.enqueue("tss");
-  });
+  std::thread producer{ getSimpleProducerThread(queue) };
+  std::thread consumer{ getInfiniteReaddingDequeueConsumerThread(queue, results) };
+  producer.join();
+  std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  queue.stop();
+  consumer.join();
+  REQUIRE(utils::StringUtils::join("-", results) == "ba-dum-tss");
+}
 
-  std::thread consumer([&queue, &results]() {
-    std::string s;
-    while (queue.dequeueWait(s)) {
-      results.insert(s);
-      queue.enqueue(std::move(s));
-    }
-  });
+TEST_CASE("TestConditionConqurrentQueue::testConditionQueueDequeueWaitForWithSignal", "[testConditionQueueDequeueWaitForWithSignal]") {
+  using namespace MinifiConcurrentQueueTestProducersConsumers;
+  utils::ConditionConcurrentQueue<std::string> queue(true);
+  std::set<std::string> results;
+
+  std::thread producer{ getSimpleProducerThread(queue) };
+  std::thread consumer{ getDequeueWaitForConsumerThread(queue, results) };
 
   producer.join();
+  consumer.join();
 
-  std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  REQUIRE(utils::StringUtils::join("-", results) == "ba-dum-tss");
+}
 
-  queue.stop();
+TEST_CASE("TestConditionConqurrentQueue::testConditionQueueDequeueApplyWaitForWithSignal", "[testConditionQueueDequeueApplyWaitForWithSignal]") {
+  using namespace MinifiConcurrentQueueTestProducersConsumers;
+  utils::ConditionConcurrentQueue<std::string> queue(true);
+  std::set<std::string> results;
 
+  std::thread producer{ getSimpleProducerThread(queue) };
+  std::thread consumer{ getDequeueWaitForConsumerThread(queue, results) };
+  producer.join();
   consumer.join();
 
   REQUIRE(utils::StringUtils::join("-", results) == "ba-dum-tss");
 }
 
+TEST_CASE("TestConditionConqurrentQueue::testConditionQueueDequeueWaitForNoSignal", "[testConditionQueueDequeueWaitForNoSignal]") {
+  using namespace MinifiConcurrentQueueTestProducersConsumers;
+  utils::ConditionConcurrentQueue<std::string> queue(true);
+  std::set<std::string> results;
+  std::mutex mutex;
+  std::unique_lock<std::mutex> lock(mutex);
+
+  std::thread producer{ getBlockedProducerThread(queue, mutex) };
+  std::thread consumer{ getDequeueWaitForConsumerThread(queue, results) };
+  consumer.join();
+  lock.unlock();
+  producer.join();
+
+  REQUIRE(0 == results.size());
+}
+
+TEST_CASE("TestConditionConqurrentQueue::testConditionQueueDequeueApplyWaitForNoSignal", "[testConditionQueueDequeueApplyWaitForNoSignal]") {
+  using namespace MinifiConcurrentQueueTestProducersConsumers;
+  utils::ConditionConcurrentQueue<std::string> queue(true);
+  std::set<std::string> results;
+  std::mutex mutex;
+  std::unique_lock<std::mutex> lock(mutex);
+
+  std::thread producer{ getBlockedProducerThread(queue, mutex) };
+  std::thread consumer{ getDequeueWaitForConsumerThread(queue, results) };
```

Comment:
There are some copy-paste errors with the wrong functions being called. Could you double-check them?

This test is identical to the above but its name suggests that it shouldn't be.

Reply:
Ah, I see, that is a very good catch, this should be calling the consumer using `apply`. Also, just realized, that the strings `TestConcurrentqueue`/`TestConditionConcurrentQueue` are consistently (but not consequently) mistyped, will correct at least that as well.

Reply:
Replaced duplicate tests and fixed typos.